### PR TITLE
feat(TypeScript): adding type definitions

### LINKED ISSUE #324: Support for Typescript
It is impossible to use this library with Typescript. It would be nice if it generated types for typescript as well, or better yet was rewritten for Typescript :D

### PR DESCRIPTION
**Pull request by RobinCsl**
These were originally created by @kravco in an internal project at Kiwi.com, and they have not been updated in a few months, so there might be some discrepancies.

TODO
- [x] Check that each type matches up against the Flow types;
- [x] Alter the build process to include these files in the shipped NPM package

ALTERNATIVE
- submit these types on DefinitelyTyped and manage it there

**Components done:**

- [x] AirportIllustration
- [x] Alert
- [x] Badge
- [x] Breadcrumbs
- [x] Button
- [x] ButtonGroup
- [x] ButtonLink
- [x] CallOutBanner
- [x] Card
- [x] CarrierLogo
- [x] Checkbox
- [x] ChoiceGroup
- [x] ClickOutside
- [x] Collapse
- [x] CountryFlag
- [x] Coupon
- [x] Desktop
- [x] Dialog
- [x] Dictionary
- [x] Drawer
- [x] FeatureIcon
- [x] FormFeedback
- [x] FormLabel
- [x] Heading
- [x] Hide
- [x] Icon
- [x] icons
- [x] Illustration
- [x] InputField
- [x] InputFile
- [x] InputGroup
- [x] InputStepper
- [x] Layout
- [x] LazyImage
- [x] LinkList
- [x] List
- [x] ListChoice
- [x] Loading
- [x] Mobile
- [x] Modal
- [x] ModalFooter
- [x] ModalHeader
- [x] ModalSection
- [x] NavigationBar
- [x] NotificationBadge
- [x] Pagination
- [x] PictureCard
- [x] Popover
- [x] Portal
- [x] PricingTable
- [x] BadgePrimitive
- [x] IllustrationPrimitive
- [x] Radio
- [x] RatingStars
- [x] Seat
- [x] Select
- [x] Separator
- [x] ServiceLogo
- [x] SkipLink
- [x] SkipNavigation
- [x] Slider
- [x] Stack
- [x] Stepper
- [x] StepperStateless
- [x] Sticky
- [x] StopoverArrow
- [x] Table
- [x] Tag
- [x] Text
- [x] Textarea
- [x] TextLink
- [x] ThemeProvider
- [x] Tile
- [x] TileGroup
- [x] Tooltip
- [x] Translate
- [x] Truncate
- [x] boundingClientRect
- [x] cloneWithTooltip
- [x] Grid
- [x] handleKeyDown
- [x] mediaQuery
- [x] randomID
- [x] rtl
- [x] scroll
- [x] Slide
- [x] toggleDown
- [x] toggleUp
- [x] transition
- [x] validateDecrement
- [x] validateIncrement

Template for icons:
- [x] FeatureIcon
- [x] Icon

Closes #324<br/><br/><br/><url>LiveURL: https://orbit-components-rcsl-add-ts-definitions.surge.sh</url>

### CODE REVIEW

**Comment by RobinCsl** [comment]
Hi @vepor,
Do you have any ETA for when this would be released?

**Comment by saharbit** [comment]
Hey, any update?

**Comment by vepor** [comment]
@saharbit Not yet, sorry, we need to sync new design updates currently 😕. We'll do our best to finish it until the end of March.

File: .eslintTS.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module.exports =  {
+  parser:  "@typescript-eslint/parser",
+  extends:  [
+    "plugin:react/recommended",
+    "plugin:@typescript-eslint/recommended",
+    "prettier/@typescript-eslint",
+    "plugin:prettier/recommended",
+    'plugin:prettier/recommended',
```
Comment (RobinCsl) [inline .eslintTS.js]: @vepor this plugin is written twice, probably not intentional

**Comment by yusufali19** [comment]
Hey, any idea when this feature will be completed? 
We are eagerly waiting for this feature!

**Comment by filipdanisko** [comment]
@yusufali19 stable release will probably be next week, until then, feel free to test it out with release candidate https://github.com/kiwicom/orbit-components/releases/tag/0.85.0-rc.1

**Comment by saharbit** [comment]
@filipdanisko thank you!
Icon types seems to be missing? or just not working as intended?

**Comment by filipdanisko** [comment]
@saharbit I will look in to that, thanks for report :)

**Comment by saharbit** [comment]
> @saharbit I will look in to that, thanks for report :) 

Let me know how I can help

**Comment by filipdanisko** [comment]
> > @saharbit I will look in to that, thanks for report :)
> 
> Let me know how I can help

I fixed the icons, it was a typo in exporting file, but can hunt the issue with callbacks, if you could give me an example of that, that would be fantastic! :)

**Comment by filipdanisko** [comment]
Also rc2 released :) https://github.com/kiwicom/orbit-components/releases/tag/0.85.0-rc.2

**Comment by JLee21** [comment]
Merged! You guys are awesome!

**Pull request merged**